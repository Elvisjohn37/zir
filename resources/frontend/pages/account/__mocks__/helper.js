let mockedModule = jest.createMockFromModule('../helper.js');

mockedModule.openAccountWindow = jest.fn();

module.exports = mockedModule;
