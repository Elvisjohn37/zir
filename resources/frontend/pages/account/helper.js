export function openAccountWindow(url) {
	window.open(url, 'account_window', 'width = 670, height = 800');
}
