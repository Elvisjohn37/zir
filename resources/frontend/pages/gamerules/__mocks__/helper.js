let mockedModule = jest.createMockFromModule('../helper');

mockedModule.extractProductGameType = (gameRulesList) => {
	let extractedItems = [];
	let indexMap = {};
	let nextIndex = 0;

	gameRulesList.forEach((item) => {
		if (item.productName in indexMap) {
			if (!extractedItems[indexMap[item.productName]].gameTypes.includes(item.gameTypeName)) {
				extractedItems[indexMap[item.productName]].gameTypes.push(item.gameTypeName);
			}
		} else {
			indexMap[item.productName] = nextIndex;
			nextIndex++;

			extractedItems.push({
				productName: item.productName,
				gameTypes: [item.gameTypeName],
			});
		}
	});

	return extractedItems;
};

module.exports = mockedModule;
