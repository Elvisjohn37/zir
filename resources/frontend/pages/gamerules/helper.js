import lodash from 'lodash';

function extractProductGameType(gameRulesList) {
	let extractedItems = [];
	let indexMap = {};
	let nextIndex = 0;

	gameRulesList.forEach((item) => {
		if (item.productName in indexMap) {
			if (!lodash.includes(extractedItems[indexMap[item.productName]].gameTypes, item.gameTypeName)) {
				extractedItems[indexMap[item.productName]].gameTypes.push(item.gameTypeName);
			}
		} else {
			indexMap[item.productName] = nextIndex;
			nextIndex++;

			extractedItems.push({
				productName: item.productName,
				gameTypes: [item.gameTypeName],
			});
		}
	});

	return extractedItems;
}

export { extractProductGameType };
