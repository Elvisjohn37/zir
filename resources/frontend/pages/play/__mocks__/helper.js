let postMessageHandler = () => jest.fn();

let setPostMessageHandler = () => jest.fn();

let resetPostMessageHandler = () => jest.fn();

resetPostMessageHandler();

// _________________________________________

let windowResponseData,
	isMobile = false;

let windowResponse = () => windowResponseData;

let resetWindowResponse = () => {
	windowResponseData = {
		result: true,
		data: {
			url: 'https://eyecon.classicku.com/launch/lek?uid=ANqWb0TXAFr1%252BZT7aQmY%252F1M%253D&guid=124275588462ff2e5f3973e3.1098258&gameid=71656&brand=LEK&cur=ZWD&lang=en&homeURL=https%3A%2F%2Fgames.classicku.com%2Fclosegame%3FgameID%3DDVbr%26clientID%3DML641',
			title: 'Beat The Bobbies',
			maxPayout: '700000000.000000',
			isMobile: false,
			viewport: 'desktop',
			productID: 'zy',
			balanceInfo: { playableBalance: '1457.800000', currencyCode: 'ZWD' },
		},
	};
};

let setWindowResponse = (data) => {
	windowResponseData = {
		...windowResponseData,
		data: {
			...windowResponseData.data,
			...data,
		},
	};
};

let setErrorResponse = () => {
	windowResponseData = {
		...windowResponseData,
		result: false,
	};
};

resetWindowResponse();

// _________________________________________

module.exports = { postMessageHandler, windowResponse, setWindowResponse, resetWindowResponse, setErrorResponse };
