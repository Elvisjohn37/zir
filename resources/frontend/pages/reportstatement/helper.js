import { getStatementReport as getStatementReportAjax } from 'frontend/ajax/backend';

export function getStatementReport(
	isMounted,
	setDateRange,
	statementRange,
	setStatementData,
	setStatementDataLoading,
	setStatementDataError
) {
	setStatementDataLoading(true);

	getStatementReportAjax(
		statementRange,
		!isMounted,
		(response) => {
			setStatementDataLoading(false);
			setStatementData({
				content: response.data.data.content,
				footer: response.data.data.footer,
			});
			if (!isMounted) {
				setDateRange({
					isLoading: false,
					range: response.data.data.dateRange,
				});
			}
		},
		() => {
			setStatementDataError(true);
			setStatementDataLoading(false);
		}
	);
}
