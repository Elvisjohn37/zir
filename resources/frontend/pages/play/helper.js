import { windowResponse } from 'frontend/utils/helper';

let postMessageHandlerWrapper = () => {
	let allowedOrigin = windowResponse() ? windowResponse().data.url : '';

	return function ({ data, origin }) {
		let originUrl = new URL(origin);
		let gameUrl = new URL(allowedOrigin);

		if ((gameUrl.origin == originUrl.origin || window.location.host == originUrl.host) && data == 'closeGame') {
			window.location.href = '/';
		}
	};
};

let postMessageHandler = postMessageHandlerWrapper();

export { postMessageHandler, windowResponse };
