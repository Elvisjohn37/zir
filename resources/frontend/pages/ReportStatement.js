import React, { useEffect } from 'react';
import styles from './ReportStatement.module';
import { LangDataProvider } from 'frontend/components/Language';
import { FormattedMessage } from 'react-intl';
import Table from '@mui/material/Table';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableFooter from '@mui/material/TableFooter';
import Sidebar from 'frontend/components/Sidebar';
import MediaQuery from 'react-responsive';
import TableIndicator from 'frontend/components/TableIndicator';
import { Date, GrossRake } from 'frontend/components/ReportCommonFormatter';
import {
	Product,
	Turnover,
	Commission,
	CashBalance,
	Credit,
	DateLink,
} from 'frontend/components/ReportStatementFormatter';
import { DesktopDateRange, MobileDateRange } from './reportstatement/DateRange';
import { getStatementReport } from './reportstatement/helper';
import { LayoutConsumer } from 'frontend/contexts/Layout';
import Spinner from 'frontend/components/Spinner';

export default function ReportStatement() {
	let [statementData, setStatementData] = React.useState({
		content: [],
		footer: [],
	});
	let [statementDataLoading, setStatementDataLoading] = React.useState(true);
	let [statementDataError, setStatementDataError] = React.useState(false);

	let [dateRange, setDateRange] = React.useState({
		isLoading: true,
		range: [],
	});

	let [isMounted, setIsMounted] = React.useState(false);

	const [activeDate, setActiveDate] = React.useState(1);

	useEffect(() => {
		setIsMounted(true);
		getStatementReport(
			isMounted,
			setDateRange,
			activeDate,
			setStatementData,
			setStatementDataLoading,
			setStatementDataError
		);
	}, []);

	const handleChange = (event, activeDateRange) => {
		getStatementReport(
			isMounted,
			setDateRange,
			activeDateRange,
			setStatementData,
			setStatementDataLoading,
			setStatementDataError
		);
		setActiveDate(activeDateRange);
	};

	return (
		<MediaQuery maxDeviceWidth={styles.main_mobile}>
			{(matches) =>
				<LayoutConsumer>
					{({ layoutState }) => (
						<LangDataProvider category="reports" fallback={<Spinner/>}>
							<Sidebar sidebarGroup="archivedReport" disabled={layoutState.config.isMobileSite}>
								<div className={styles.tableSection}>
									<div className={styles.dates}>
										{matches ? (
											<MobileDateRange dateRange={dateRange} onChange={handleChange} activeDate={activeDate} />
										) : (
											<DesktopDateRange dateRange={dateRange} onChange={handleChange} activeDate={activeDate} />
										)}
									</div>

									<div className={styles.tableContent}>
										<TableIndicator
											isLoading={statementDataLoading}
											isEmpty={statementData.content.length == 0}
											isError={statementDataError}
										>
											<TableContainer>
												<Table>
													<TableHead>
														<TableRow classes={{ root: styles.row }}>
															<TableCell align="left" className={styles.colHeaderRowNum}>
																#
															</TableCell>
															<TableCell align="center" className={styles.colHeaderDate}>
																<FormattedMessage id="date" />
															</TableCell>
															<TableCell align="center" className={styles.colHeaderType}>
																<FormattedMessage id="type" />
															</TableCell>
															<TableCell align="center" className={styles.colHeaderProduct}>
																<FormattedMessage id="product" />
															</TableCell>
															<TableCell align="right" className={styles.colHeaderTurnover}>
																<FormattedMessage id="turnover" />
															</TableCell>
															<TableCell align="right" className={styles.colHeaderGrossRake}>
																<FormattedMessage id="grossRake" />
															</TableCell>
															<TableCell align="right" className={styles.colHeaderCommission}>
																<FormattedMessage id="commission" />
															</TableCell>
															<TableCell align="right" className={styles.colHeaderCashBalance}>
																<FormattedMessage id="cashBalance" />
															</TableCell>
															<TableCell align="right" className={styles.colHeaderCredit}>
																<FormattedMessage id="credit" />
															</TableCell>
														</TableRow>
													</TableHead>

													<TableBody>
														{statementData.content.map((row) => (
															<TableRow key={row.rowNo}>
																<TableCell align="left" scope="row">
																	{row.rowNo}
																</TableCell>
																<TableCell align="center">
																	<DateLink row={row}>
																		<Date date={row.date} />
																	</DateLink>
																</TableCell>
																<TableCell align="center">{row.type}</TableCell>
																<TableCell align="center">
																	<Product row={row} />
																</TableCell>
																<TableCell align="right">
																	<Turnover row={row} />
																</TableCell>
																<TableCell align="right">
																	<GrossRake row={row} />
																</TableCell>
																<TableCell align="right">
																	<Commission row={row} />
																</TableCell>
																<TableCell align="right">
																	<CashBalance row={row} />
																</TableCell>
																<TableCell align="right">
																	<Credit row={row} />
																</TableCell>
															</TableRow>
														))}
													</TableBody>
													<TableFooter>
														<TableRow>
															<TableCell colSpan={3} />
															<TableCell align="right">
																<FormattedMessage id="total" />
															</TableCell>
															<TableCell align="right">
																<Turnover row={statementData.footer} />
															</TableCell>
															<TableCell align="right">
																<GrossRake row={statementData.footer} />
															</TableCell>
															<TableCell align="right">
																<Commission row={statementData.footer} />
															</TableCell>
															<TableCell align="right">
																<CashBalance row={statementData.footer} />
															</TableCell>
															<TableCell align="right">
																<Credit row={statementData.footer} />
															</TableCell>
														</TableRow>
													</TableFooter>
												</Table>
											</TableContainer>
										</TableIndicator>
									</div>
								</div>
							</Sidebar>
						</LangDataProvider>
					)}
				</LayoutConsumer>
			}
		</MediaQuery>
	);
}
