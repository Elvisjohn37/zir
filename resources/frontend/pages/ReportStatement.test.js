import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import ReportStatement from './ReportStatement';
import * as StatementReportHelper from './reportstatement/helper';
import * as MediaQuery from 'react-responsive';
import userEvent from '@testing-library/user-event';
import { setUserState } from 'frontend/contexts/User';

jest.mock('@mui/material/TableRow');
jest.mock('@mui/material/TableBody');
jest.mock('@mui/material/TableCell');
jest.mock('@mui/material/TableFooter');
jest.mock('./reportstatement/helper');
jest.mock('./reportstatement/DateRange');
jest.mock('frontend/components/Language');
jest.mock('react-intl');
jest.mock('frontend/components/Sidebar');
jest.mock('react-responsive');
jest.mock('frontend/components/TableIndicator');
jest.mock('frontend/components/ReportCommonFormatter');
jest.mock('frontend/components/ReportStatementFormatter');
jest.mock('frontend/contexts/Layout');
jest.mock('frontend/contexts/User');

describe('</ReportStatement>', () => {
	let rsAsFragment, rsContainer, rsRerender, rsGetByText;

	beforeEach(() => {
		jest.spyOn(StatementReportHelper, 'getStatementReport');
		jest.spyOn(MediaQuery, 'setMediaQuery');
		let { asFragment, container, rerender, getByText } = renderReportStatement();
		rsAsFragment = asFragment;
		rsContainer = container;
		rsRerender = rerender;
		rsGetByText = getByText;
	});

	afterEach(() => {
		resetMocks();
		cleanup();
		rsAsFragment = null;
		rsContainer = null;
		rsRerender = null;
		rsGetByText = null;
	});

	function resetMocks() {
		jest.clearAllMocks();
		jest.restoreAllMocks();
		StatementReportHelper.resetGetStatementReport();
		MediaQuery.setMediaQuery(true);
	}

	function renderReportStatement() {
		let { asFragment, container, rerender, getByText } = render(<ReportStatement />);
		return { asFragment, container, rerender, getByText };
	}

	function expectCorrectTableContent() {
		setUserState({ user: { parent: 'test' } });
		let responseData = StatementReportHelper.getStatementReportData().data.data;
		let columnsLength = 9;
		responseData.content.forEach((data, index) => {
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 1]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 2]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 3]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 4]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 5]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 6]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 7]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 8]).toBeInTheDocument();
		});
	}

	it('should match snapshot', () => {
		expect(rsAsFragment()).toMatchSnapshot();
	});

	it('should display loader when initiating the page', () => {
		resetMocks();
		StatementReportHelper.setIsProceedHook(false);
		let { getByText } = renderReportStatement();
		expect(getByText('Loading TableIndicator')).toBeInTheDocument();
	});

	it('should display date picker base on device type', () => {
		expect(screen.getByTestId('MobileDateRange')).toBeInTheDocument();
		resetMocks();
		cleanup();
		MediaQuery.setMediaQuery(false);
		render(<ReportStatement />);
		expect(screen.getByTestId('DesktopDateRange')).toBeInTheDocument();
	});

	it('should get record only from backend and display it when player pick a date', () => {
		userEvent.click(screen.getByTestId('MobileDateRange'));
		expectCorrectTableContent();
	});

	it('should display no record found when record is empty', () => {
		cleanup();
		StatementReportHelper.setGetStatementReport(true, { content: [] });
		let { container } = renderReportStatement();
		let tableRow = container.getElementsByClassName('testTableRow');
		expect(tableRow).toHaveLength(0);
	});

	it('should display error message when backend response is not valid', () => {
		cleanup();
		StatementReportHelper.setGetStatementReport(false);
		renderReportStatement();
		expect(screen.getByText('Error Response')).toBeInTheDocument();
	});
});
