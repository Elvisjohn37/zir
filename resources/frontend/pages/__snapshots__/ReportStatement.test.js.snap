// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</ReportStatement> should match snapshot 1`] = `
<DocumentFragment>
  <div
    category="reports"
    data-testid="LangDataProvider"
  >
    <div>
      reports
    </div>
    <div
      data-testid="default"
      sidebargroup="report"
    >
      <div
        class="tableSection"
      >
        <div
          class="dates"
        >
          <div
            activedate="1"
            data-testid="MobileDateRange"
            daterange="{\\"isLoading\\":false,\\"range\\":[{\\"number\\":3,\\"startDate\\":\\"2021-07-01\\",\\"endDate\\":\\"2021-07-31\\"},{\\"number\\":2,\\"startDate\\":\\"2021-08-01\\",\\"endDate\\":\\"2021-08-31\\"},{\\"number\\":1,\\"startDate\\":\\"2021-09-01\\",\\"endDate\\":\\"2021-09-01\\"}]}"
          >
            <span>
              activeDate: 1
            </span>
            <span>
              row: 
            </span>
          </div>
        </div>
        <div
          class="tableContent"
        >
          <div
            data-testid="TableIndicator"
            isempty="false"
            iserror="false"
            isloading="false"
          >
            <span>
              Done loading TableIndicator
            </span>
            <div
              data-testid="TableContainer"
            >
              <div
                data-testid="Table"
              >
                <div
                  data-testid="TableHead"
                >
                  <div
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="left"
                      class="colHeaderRowNum"
                      data-testid="TableCell"
                    >
                      #
                    </div>
                    <div
                      align="center"
                      class="colHeaderDate"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="date"
                      >
                        <div>
                          date
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="colHeaderType"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="type"
                      >
                        <div>
                          type
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="colHeaderProduct"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="product"
                      >
                        <div>
                          product
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="colHeaderTurnover"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="turnover"
                      >
                        <div>
                          turnover
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="colHeaderGrossRake"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="grossRake"
                      >
                        <div>
                          grossRake
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="colHeaderCommission"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="commission"
                      >
                        <div>
                          commission
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="colHeaderCashBalance"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="cashBalance"
                      >
                        <div>
                          cashBalance
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="colHeaderCredit"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="credit"
                      >
                        <div>
                          credit
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  data-testid="TableBody"
                >
                  <div
                    data-testid="TableRow"
                  >
                    <div
                      align="left"
                      data-testid="TableCell"
                      scope="row"
                    >
                      1
                    </div>
                    <div
                      align="center"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateLink"
                        row="{\\"rowNo\\":1,\\"type\\":\\"Betting\\",\\"date\\":\\"2021-09-01\\",\\"productFormat\\":\\"defaultProductName\\",\\"productID\\":\\"zy\\",\\"productName\\":\\"Games\\",\\"turnover\\":\\"101.200000\\",\\"grossRake\\":\\"0.000000\\",\\"commission\\":\\"0.006000\\",\\"cashBalance\\":\\"-100.500000\\",\\"credit\\":\\"0.000000\\"}"
                      >
                        <div
                          data-testid="Date"
                          date="2021-09-01"
                        />
                      </div>
                    </div>
                    <div
                      align="center"
                      data-testid="TableCell"
                    >
                      Betting
                    </div>
                    <div
                      align="center"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="Product"
                      >
                        Product Name: Default
                      </div>
                    </div>
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="Turnover"
                        row="{\\"rowNo\\":1,\\"type\\":\\"Betting\\",\\"date\\":\\"2021-09-01\\",\\"productFormat\\":\\"defaultProductName\\",\\"productID\\":\\"zy\\",\\"productName\\":\\"Games\\",\\"turnover\\":\\"101.200000\\",\\"grossRake\\":\\"0.000000\\",\\"commission\\":\\"0.006000\\",\\"cashBalance\\":\\"-100.500000\\",\\"credit\\":\\"0.000000\\"}"
                      />
                    </div>
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="GrossRake"
                      >
                        <div>
                          {"rowNo":1,"type":"Betting","date":"2021-09-01","productFormat":"defaultProductName","productID":"zy","productName":"Games","turnover":"101.200000","grossRake":"0.000000","commission":"0.006000","cashBalance":"-100.500000","credit":"0.000000"}
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="Commission"
                        row="{\\"rowNo\\":1,\\"type\\":\\"Betting\\",\\"date\\":\\"2021-09-01\\",\\"productFormat\\":\\"defaultProductName\\",\\"productID\\":\\"zy\\",\\"productName\\":\\"Games\\",\\"turnover\\":\\"101.200000\\",\\"grossRake\\":\\"0.000000\\",\\"commission\\":\\"0.006000\\",\\"cashBalance\\":\\"-100.500000\\",\\"credit\\":\\"0.000000\\"}"
                      />
                    </div>
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="CashBalance"
                        row="{\\"rowNo\\":1,\\"type\\":\\"Betting\\",\\"date\\":\\"2021-09-01\\",\\"productFormat\\":\\"defaultProductName\\",\\"productID\\":\\"zy\\",\\"productName\\":\\"Games\\",\\"turnover\\":\\"101.200000\\",\\"grossRake\\":\\"0.000000\\",\\"commission\\":\\"0.006000\\",\\"cashBalance\\":\\"-100.500000\\",\\"credit\\":\\"0.000000\\"}"
                      />
                    </div>
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="Credit"
                        row="{\\"rowNo\\":1,\\"type\\":\\"Betting\\",\\"date\\":\\"2021-09-01\\",\\"productFormat\\":\\"defaultProductName\\",\\"productID\\":\\"zy\\",\\"productName\\":\\"Games\\",\\"turnover\\":\\"101.200000\\",\\"grossRake\\":\\"0.000000\\",\\"commission\\":\\"0.006000\\",\\"cashBalance\\":\\"-100.500000\\",\\"credit\\":\\"0.000000\\"}"
                      />
                    </div>
                  </div>
                  <div
                    data-testid="TableRow"
                  >
                    <div
                      align="left"
                      data-testid="TableCell"
                      scope="row"
                    >
                      2
                    </div>
                    <div
                      align="center"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateLink"
                        row="{\\"rowNo\\":2,\\"type\\":\\"Transfer\\",\\"date\\":\\"2021-09-01\\",\\"productFormat\\":\\"defaultProductName\\",\\"productID\\":\\"8q\\",\\"productName\\":\\"\\",\\"turnover\\":\\"0.000000\\",\\"grossRake\\":\\"0.000000\\",\\"commission\\":\\"0.000000\\",\\"cashBalance\\":\\"0.000000\\",\\"credit\\":\\"0.000000\\"}"
                      >
                        <div
                          data-testid="Date"
                          date="2021-09-01"
                        />
                      </div>
                    </div>
                    <div
                      align="center"
                      data-testid="TableCell"
                    >
                      Transfer
                    </div>
                    <div
                      align="center"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="Product"
                      >
                        Product Name: Default
                      </div>
                    </div>
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="Turnover"
                        row="{\\"rowNo\\":2,\\"type\\":\\"Transfer\\",\\"date\\":\\"2021-09-01\\",\\"productFormat\\":\\"defaultProductName\\",\\"productID\\":\\"8q\\",\\"productName\\":\\"\\",\\"turnover\\":\\"0.000000\\",\\"grossRake\\":\\"0.000000\\",\\"commission\\":\\"0.000000\\",\\"cashBalance\\":\\"0.000000\\",\\"credit\\":\\"0.000000\\"}"
                      />
                    </div>
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="GrossRake"
                      >
                        <div>
                          {"rowNo":2,"type":"Transfer","date":"2021-09-01","productFormat":"defaultProductName","productID":"8q","productName":"","turnover":"0.000000","grossRake":"0.000000","commission":"0.000000","cashBalance":"0.000000","credit":"0.000000"}
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="Commission"
                        row="{\\"rowNo\\":2,\\"type\\":\\"Transfer\\",\\"date\\":\\"2021-09-01\\",\\"productFormat\\":\\"defaultProductName\\",\\"productID\\":\\"8q\\",\\"productName\\":\\"\\",\\"turnover\\":\\"0.000000\\",\\"grossRake\\":\\"0.000000\\",\\"commission\\":\\"0.000000\\",\\"cashBalance\\":\\"0.000000\\",\\"credit\\":\\"0.000000\\"}"
                      />
                    </div>
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="CashBalance"
                        row="{\\"rowNo\\":2,\\"type\\":\\"Transfer\\",\\"date\\":\\"2021-09-01\\",\\"productFormat\\":\\"defaultProductName\\",\\"productID\\":\\"8q\\",\\"productName\\":\\"\\",\\"turnover\\":\\"0.000000\\",\\"grossRake\\":\\"0.000000\\",\\"commission\\":\\"0.000000\\",\\"cashBalance\\":\\"0.000000\\",\\"credit\\":\\"0.000000\\"}"
                      />
                    </div>
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="Credit"
                        row="{\\"rowNo\\":2,\\"type\\":\\"Transfer\\",\\"date\\":\\"2021-09-01\\",\\"productFormat\\":\\"defaultProductName\\",\\"productID\\":\\"8q\\",\\"productName\\":\\"\\",\\"turnover\\":\\"0.000000\\",\\"grossRake\\":\\"0.000000\\",\\"commission\\":\\"0.000000\\",\\"cashBalance\\":\\"0.000000\\",\\"credit\\":\\"0.000000\\"}"
                      />
                    </div>
                  </div>
                </div>
                <div
                  data-testid="TableFooter"
                >
                  <div
                    data-testid="TableRow"
                  >
                    <div
                      colspan="3"
                      data-testid="TableCell"
                    />
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="total"
                      >
                        <div>
                          total
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="Turnover"
                        row="{\\"grossRake\\":0,\\"turnover\\":101.2,\\"commission\\":0.006,\\"cashBalance\\":-100.5,\\"credit\\":0}"
                      />
                    </div>
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="GrossRake"
                      >
                        <div>
                          {"grossRake":0,"turnover":101.2,"commission":0.006,"cashBalance":-100.5,"credit":0}
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="Commission"
                        row="{\\"grossRake\\":0,\\"turnover\\":101.2,\\"commission\\":0.006,\\"cashBalance\\":-100.5,\\"credit\\":0}"
                      />
                    </div>
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="CashBalance"
                        row="{\\"grossRake\\":0,\\"turnover\\":101.2,\\"commission\\":0.006,\\"cashBalance\\":-100.5,\\"credit\\":0}"
                      />
                    </div>
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="Credit"
                        row="{\\"grossRake\\":0,\\"turnover\\":101.2,\\"commission\\":0.006,\\"cashBalance\\":-100.5,\\"credit\\":0}"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
